Add ValidMind Public REST API to docs site

nrichers opened the pull request:
# Pull Request Description

## What and why?
<!-- Briefly describe what changed and why — include behavior before and after if helpful -->
<!-- Frontend: Add screenshots or videos showing before/after -->

This PR integrates our new REST APIs by adding the following:  

- **ValidMind Public REST API reference**
   - A new landing page with overview information
   - Rendered Swagger JSON for the public API
- **ValidMind Admin API** <<< MOVED TO [#121](https://github.com/validmind/installation/pull/121) 

Under the covers, we use [RapiDoc](https://github.com/rapi-doc/RapiDoc), which is MIT licensed, to render the API reference content directly from the published JSON specs.

For inspecting the JSON and RapiDoc JavaScript locally, there is also a new `make get-api-json` action that fetches related assets. 

Rel.: [sc-11108](https://app.shortcut.com/validmind/story/11108/documentation-pipeline-for-api-publishing)

## How to test
<!-- Describe how the change has been tested, and how a reviewer can test the change locally -->

If you want the cURL examples to include the complete header, make sure to add some authentication info:

<img width="879" height="626" alt="Capto_ 2025-07-29_03-35-36_pm" src="https://github.com/user-attachments/assets/83484875-175e-4830-9455-a902064ddeea" />


### ValidMind Public REST API landing page: [**LIVE PREVIEW**](https://docs-staging.validmind.ai/pr_previews/nrichers/sc-11108/documentation-pipeline-for-api-publishing/reference/validmind-rest-api-vm.html)

<img width="1234" height="2087" alt="Capto_ 2025-07-31_03-07-40_pm" src="https://github.com/user-attachments/assets/6aa6b65b-21d1-4cdc-8137-3246b02164df" />

### ValidMind Public API reference: [**LIVE PREVIEW**](https://docs-staging.validmind.ai/pr_previews/nrichers/sc-11108/documentation-pipeline-for-api-publishing/reference/vm-api.html) 

Sample endpoint:<br><img width="913" height="1446" alt="image" src="https://github.com/user-attachments/assets/8b44a98b-cfb6-4382-8740-d24bb1619c51" />

## What needs special review?
<!-- Optional: Call out specific areas needing detailed review -->

### Missing request content

There are some endpoints that do not seem to include methods, for example [`GET /health`](https://docs-staging.validmind.ai/pr_previews/nrichers/sc-11108/documentation-pipeline-for-api-publishing/reference/vm-api.html#get-/health). 

<img width="913" height="613" alt="Capto_ 2025-07-29_03-51-57_pm" src="https://github.com/user-attachments/assets/e8996c34-c8c6-4b77-ba6e-612633cccb1c" />

### 'Try it' feature

There is a 'try it' feature which I've enabled for this review that requires a server. If we can provide one, then the API reference could include cURL examples that even the correct headers (if you specify the a:

<img width="1083" height="835" alt="Capto_ 2025-07-29_03-18-26_pm" src="https://github.com/user-attachments/assets/1eae4361-b275-45a4-8340-34e7abdab155" />

If we cannot provide a server, we can disable the feature with `allow-try="false"`.

### Look & feel

There is some follow-on work that will need to happen to make this API reference look and feel like other content, but we should track this in a follow-on story as the current story is for the documentation pipeline. Specifically, we should update:

- Some of the RapiDoc default colors
- Some of the default whitespace and text alignments

## Dependencies, breaking changes, and deployment notes
<!-- A list of links to pull requests that are depend on or are dependencies of this PR -->
<!-- Any special deployment considerations? -->
<!-- List any breaking changes -->

N/A

## Release notes
<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

We added a new ValidMind Public REST API to our developer content: [ValidMind Rest APIs](https://docs.validmind.com/reference/validmind-rest-api-vm.html)

This API enables you to programmatically interact with models, groups, business units, use cases, and templates.

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `breaking-change`
- `deprecation`
- `documentation`
- `environment-variables`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## Checklist
- [x] What and why
- [x] Screenshots or videos (Frontend)
- [x] How to test
- [x] What needs special review
- [ ] Dependencies, breaking changes, and deployment notes
- [x] Labels applied
- [x] PR linked to Shortcut
- [ ] Unit tests added (Backend)
- [x] Tested locally
- [x] Documentation updated (if required)
- [ ] Environment variable additions/changes documented (if required)

nibalizer commented:
Key capabilities

    Model inventory management
    Retrieve, filter, create, and update models.
    Supports querying by tiering, owners, validators, status, and more.
    Supports exporting model data as JSON or CSV.

    Group and access control
    List, create, and manage access groups.
    Add or remove users and resources to/from groups.
    Filter by default groups and apply sorting.

    Business unit management
    Retrieve and manage business units associated with the organization.
    Supports filtering, search, and pagination.

    Template access
    Retrieve available templates used for model documentation and validation.
    Includes support for fetching template metadata and default structures.

    Use case management
    Create and organize use cases and use case categories.
    Supports sorting, filtering, and hierarchical structures.

    Field masking for optimized responses
    Use the X-Fields header to control which fields are returned in API responses.
    Useful for minimizing payload size and tailoring data for frontend consumption.

    Health and authentication checks
    Basic endpoints to validate authentication (/test-auth) and check service status (/health).



I think we only have:

Create Model

Not sure why it says we have stuff we don't have?

nibalizer reviewed (changes requested):
Noted in comment

cachafla commented:
On the authentication section I would provide more context/explanation about the authentication headers. The user should know that the format of that snippet, as shown, is meant for a `curl` command. The authentication should explain briefly the two auth headers required and how they can be passed (in this example to a `curl` command).

<img width="782" height="215" alt="image" src="https://github.com/user-attachments/assets/20921e98-6a0b-4ebf-8317-c500b7f7e233" />

mehdiesm0501-vm commented:
@nibalizer  I can see the preview of the API reference link: https://docs-staging.validmind.ai/pr_previews/nrichers/sc-11108/documentation-pipeline-for-api-publishing/reference/vm-api.html#post-/models

I see a lot more than Create Models documented - is that not right?

nrichers commented:
> Key capabilities

>@nrichers please confirm which API endpoints are actually currently available (per Spencer's comment). The doc and reference mention a lot of functions around use cases, groups, BUs, etc. Just want to make sure this hasn't been hallucinated!!

@nibalizer and @mehdiesm0501-vm I simplified and corrected the key capabilities listed in each API description. The content now matches the major sections of the Swagger JSON specs:

**Public API**

| Swagger JSON spec       | Static API description    |
|-------------------------|---------------------------|
| <img width="221" height="1163" alt="Capto_ 2025-07-31_09-08-58_am2" src="https://github.com/user-attachments/assets/cff67b10-6f0e-4723-9342-cf018c554194" /> | <img width="357" height="202" alt="Capto_ 2025-07-31_09-18-45_am2" src="https://github.com/user-attachments/assets/c5c74af0-1e0d-4e46-8727-a0097111c967" /> |

**Admin API**

| Swagger JSON spec       | Static API description    |
|-------------------------|---------------------------|
| <img width="221" height="733" alt="Capto_ 2025-07-31_09-11-05_am" src="https://github.com/user-attachments/assets/c453946c-58ce-4a87-834b-ec63ef7fa8bd" /> | <img width="332" height="154" alt="Capto_ 2025-07-31_09-20-01_am" src="https://github.com/user-attachments/assets/5a64396c-e60c-4291-a1e5-fdea85368323" /> |

nrichers commented:
@nibalizer I removed the admin API from this PR, leaving only the public API:

<img width="1278" height="1846" alt="Capto_ 2025-07-31_02-49-22_pm" src="https://github.com/user-attachments/assets/4e9727cc-4701-4f61-8737-a2da6c8ed771" />

If this is acceptable for now, can you approve this PR? I will post a second PR to add the admin API to the CMVM docs shortly.

nrichers commented on .github/workflows/validate-docs-site.yaml:
```diff
@@ -22,6 +22,7 @@ jobs:
       with:
         repository: validmind/installation
         path: site/_source/installation
+        ref: nrichers/sc-11108/documentation-pipeline-for-api-publishing
```
TO DO: This branch `ref` needs to be removed before merging. It serves only to provide a preview for [Add admin API to CMVM docs #121](https://github.com/validmind/installation/pull/121)
